Issue found by new nilness vet check

Playing with the new `vet`, I found the following issue:

```
$ vet ./...
/Users/dgryski/go/src/github.com/dgraph-io/dgraph/query/query.go:544:16: impossible condition: nil != nil
```

The line is https://github.com/dgraph-io/dgraph/blob/master/query/query.go#L544 , and `facetsNode` has not been assigned a value between declaration on line 384 and the if test, so it's always nil.

--- a/query/query.go
+++ b/query/query.go
@@ -380,7 +380,6 @@ func (sg *SubGraph) preTraverse(uid uint64, dst outputNode) error {
 	}
 
 	var invalidUids map[uint64]bool
-	var facetsNode outputNode
 	// We go through all predicate children of the subprotos.
 	for _, pc := range sg.Children {
 		if pc.Params.ignoreResult {
@@ -539,9 +538,6 @@ func (sg *SubGraph) preTraverse(uid uint64, dst outputNode) error {
 	if sg.Params.IgnoreReflex {
 		// Lets pop the stack.
 		sg.Params.parentIds = (sg.Params.parentIds)[:len(sg.Params.parentIds)-1]
-	}
-	if facetsNode != nil && !facetsNode.IsEmpty() {
-		dst.AddMapChild("@facets", facetsNode, false)
 	}
 
 	// Only for shortest path query we wan't to return uid always if there is

